### PR TITLE
fix(folder settings): restrict input fields to max length

### PR DESCRIPTION
**Pull request by seaerchin**
## Problem
previously, the save wasnt disabled when the state of the field was invalid. 

## Solution
1. restrict the 2 fields to their respective max length, so that they cannot exceed the given length
2. for permalink, we already transform every invalid input to `-` so we know that the field's value will always be valid
 
because of this, we dont need to disable save as the permalink and title will always be of a valid length and the permalink will consist only of alphnumeric characters and `-`

## Screenshots

https://github.com/user-attachments/assets/9631c306-571a-423b-8104-5e413fb30328

### CODE REVIEW

File: apps/studio/src/features/dashboard/components/FolderSettingsModal/FolderSettingsModal.tsx [hunk 1]
```diff
@@ -158,7 +159,12 @@ const SuspendableModalContent = ({
                     placeholder="This is a url for your folder"
                     {...field}
```
Comment (dcshzj) [inline apps/studio/src/features/dashboard/components/FolderSettingsModal/FolderSettingsModal.tsx:160]: issue: should we also add the maxLength here?

Comment (seaerchin) [reply apps/studio/src/features/dashboard/components/FolderSettingsModal/FolderSettingsModal.tsx:160]: actually i tried this before, but having the controller and the maxlength on the input doesn't quite work oddly enough, which is whyi had to slice

Comment (seaerchin) [reply apps/studio/src/features/dashboard/components/FolderSettingsModal/FolderSettingsModal.tsx:160]: wtf i just retried and it works. idk what changd from last time vs now but ok ill do the max length and remove the slice

File: apps/studio/src/features/dashboard/components/FolderSettingsModal/FolderSettingsModal.tsx [hunk 2]
```diff
@@ -158,13 +158,9 @@ const SuspendableModalContent = ({
...
+                    maxLength={MAX_FOLDER_PERMALINK_LENGTH}
                     onChange={(e) => {
-                      onChange(
-                        generateResourceUrl(e.target.value).slice(
-                          0,
-                          MAX_FOLDER_PERMALINK_LENGTH,
-                        ),
-                      )
+                      onChange(generateResourceUrl(e.target.value))
```
Comment (dcshzj) [inline apps/studio/src/features/dashboard/components/FolderSettingsModal/FolderSettingsModal.tsx:163]: issue: wait no, we still need the slice (either here or in generateResourceUrl) cos the changes can be coming from outside of the user input in this field.

Comment (seaerchin) [reply apps/studio/src/features/dashboard/components/FolderSettingsModal/FolderSettingsModal.tsx:163]: how can the change come from uotside user input ah?

Comment (dcshzj) [reply apps/studio/src/features/dashboard/components/FolderSettingsModal/FolderSettingsModal.tsx:163]: My bad, got confused, so this is fine.

**Review by dcshzj — APPROVED** [review]

**Pull request merged**